### PR TITLE
Add OpInfo for torch.histc

### PR DESCRIPTION
**Pull request by saketh-are**

### CODE REVIEW

**Review by mruberry — COMMENTED** [review]
Cool stuff, @saketh-are! Two inline comments for your review

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -2592,6 +2592,24 @@ def sample_inputs_histogramdd(op_info, device, dtype, requires_grad):
...
+
+    sample_inputs = []
+    for size, min, max in product(sizes, [0, -10], [0, 10]):
+        input_tensor = make_arg(size)
+
+        sample_inputs.append(SampleInput(input_tensor,
+                                         kwargs=dict(min=min, max=max)))
+
+        for bins in range(1, 5):
+            sample_inputs.append(SampleInput(input_tensor,
```
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py]: I think it'd be cool but it seems tricky to the point I haven't thought much about it. Maybe we could combine it with an existing test to reduce/eliminate perf concerns? I just added a test, test_noncontiguous_samples, as part of the hackathon and if you rebase you should pick it up. It does a pretty good job of detecting this (although isn't designed explicitly to do so and has gaps). 

Definitely open to new ideas for how to do this better!

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -10122,6 +10140,16 @@ def ref_pairwise_distance(input1, input2):
...
            )),
+    OpInfo('histc',
+           dtypes=floating_types(),
+           dtypesIfCUDA=floating_types_and(torch.int8, torch.int16, torch.int32, torch.int64),
+           sample_inputs_func=sample_inputs_histc,
+           supports_out=True,
+           supports_autograd=False,
+           skips=(
+               # CUDA histc returns a float tensor but does not correctly warn when passed an integral out tensor
+               DecorateInfo(unittest.skip("Skipped!"), 'TestCommon', 'test_out'),
```
[the remark this answers is not in the document]
Comment (saketh-are) [reply torch/testing/_internal/common_methods_invocations.py]: `test_out` checks a few different things including resizing non-empty out (which was fixed in this PR) but also unsafe casts (still broken).

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -10598,6 +10616,18 @@ def ref_pairwise_distance(input1, input2):
...
+           dtypes=floating_types(),
+           dtypesIfCUDA=floating_types_and(torch.int8, torch.int16, torch.int32, torch.int64),
+           sample_inputs_func=sample_inputs_histc,
+           supports_out=True,
+           supports_autograd=False,
+           skips=(
+               # CUDA histc returns a float tensor but does not correctly warn when passed an integral out tensor
+               # "AssertionError: RuntimeError not raised : Expected RuntimeError when doing an unsafe cast
+               # from a result of dtype torch.float32 into an out= with dtype torch.long"
+               DecorateInfo(unittest.skip("Expected failure!"), 'TestCommon', 'test_out'),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:10629]: Just swap this decorator for `unittest.expectedFailure` like this:

https://github.com/pytorch/pytorch/blob/a6c0edff1a60e299d3f9794407f0a1c9a269fd19/torch/testing/_internal/common_methods_invocations.py#L7856

**Review by mruberry — APPROVED** [review]
Cool; just see the expected failure update

**Pull request closed without merging**